File: AccountingSoftwareApi/Models/ApplicationUserModel.cs
using ASDataManager.Library.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AccountingSoftwareApi.Models
{
    public class ApplicationUserModel
    {
        public string Id { get; set; }
        public string Username { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string EmailAddress { get; set; }
        public string CreatedDate { get; set; } = DateTime.Now.ToString();
        public RoleModel[] Roles { get; set; } = new RoleModel[]{};
    }
}

File: ASWebUI-Angular-CLI/src/app/auth/Models/ApplicationUser.model.ts
import { Role } from "./Role.model";

export class ApplicationUser{
    id: string;
    userName: string;
    firstName: string;
    lastName: string;
    emailAddress: string;
    password: string;
    userRoles: Role[]

    constructor({
        id="",
        userName = "", 
        firstName = "",
        lastName = "",
        emailAddress = "",
        password = "",
        userRoles = [],
    }={}){
        this.id = id ;
        this.userName = userName ;
        this.firstName = firstName ;
        this.lastName = lastName ;
        this.emailAddress = emailAddress ;
        this.password = password ; 
        this.userRoles = userRoles ;
    }
}